crates/client/src/config.rs

```diff
@@ -85,6 +85,15 @@ pub struct Config {
     /// `$CACHE_DIR` is the platform-specific cache directory.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub content_dir: Option<PathBuf>,
+
+    /// The path to the directory where package content is stored.
+    ///
+    /// This path is expected to be relative to the configuration file.
+    ///
+    /// If `None`, the default of `$CACHE_DIR/warg/checkpoint` is used, where
+    /// `$CACHE_DIR` is the platform-specific cache directory.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub checkpoint_path: Option<PathBuf>,
```

Comment:
I don't think this is correct. The checkpoint should be stored per-registry, and this appears to be a single checkpoint for the client regardless of registry.

Perhaps `packages_dir` should become `registries_dir` (`$CACHE_DIR/warg/registries`), which underneath it has:

* `packages` - a directory containing the package logs
* `checkpoint` - a file containing the last known registry checkpoint
* `operator.log` - the registry's operator log (for the future, don't need to implement this now as we're not doing anything with the operator log currently).

Reply:
Ah this resonates.  I was actually looking into this [client operator issue](https://github.com/bytecodealliance/registry/issues/83) a bit this morning, and was thinking it would require something like what you're suggesting.  I'll try and have some changes up soon.